DryIoc/Container.cs

```diff
@@ -417,6 +417,21 @@ private void ThrowIfContainerDisposed()
                 Throw.It(Error.ContainerIsDisposed, ToString());
         }
 
+        /// <inheritdoc />
+        public object New(Type concreteType, Made made = null,
+            RegistrySharing registrySharing = RegistrySharing.CloneButKeepCache)
+        {
+            var containerClone = With(Rules, ScopeContext,

```

Comment:
From the code alone `New` heavily uses `IContainer`, `IRegistrator` and `IResolver` facilities. 
Therefore it is hard to put into just `IResolver` or `IResolverContext` buckets.

Comparing to the when I recently made available `RegisterMapping`, `RegisterPlacholder` and `UseInstance` on `IRegistrator`, the problem here is much more scattered functionality,

Reply:
Moving this way of this PR we may consider sharing methods between all these interfaces, effectively ending up with `IContainer` as a most common denominator. So we also should return `IContainer` from the methods as well.
Maybe it is not so bad idea, but the interfaces also are used in generated code (currently it is `IResolverContext`). So making `OpenScope` return `IContainer` makes code generating much harder. Sorry if it is not so clear from my explanation. Look what interfaces are implemented in `DryIocZero` for the reference.